src/dappsExplorer/searchFilter/FavoriteDappsSection.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { useEffect } from 'react'
+import { useTranslation } from 'react-i18next'
+import { StyleSheet, Text, View } from 'react-native'
+import { useSelector } from 'react-redux'
+import { favoriteDappsWithCategoryNamesSelector } from 'src/dapps/selectors'
+import { ActiveDapp, DappSection, DappV2 } from 'src/dapps/types'
+import DappCard from 'src/dappsExplorer/DappCard'
+import { searchDappList } from 'src/dappsExplorer/searchDappList'
+import NoResults from 'src/dappsExplorer/searchFilter/NoResults'
+import StarIllustration from 'src/icons/StarIllustration'
+import Colors from 'src/styles/colors'
+import fontStyles from 'src/styles/fonts'
+import { Spacing } from 'src/styles/styles'
+
+interface Props {
+  filterId: string
+  filterName: string
+  onPressDapp: (dapp: ActiveDapp) => void
+  onShowSearchResult: (empty: boolean) => void
+  removeFilter: () => void
+  searchTerm: string
+}
+
+export function FavoriteDappsSection({
+  filterId,
+  filterName,
+  onPressDapp,
+  onShowSearchResult,
+  removeFilter,
+  searchTerm,
+}: Props) {
+  const { t } = useTranslation()
+  const favoriteDappsWithCategoryNames = useSelector(favoriteDappsWithCategoryNamesSelector)
+  const favoriteResultsFiltered = favoriteDappsWithCategoryNames.filter(
+    (dapp) => dapp.categories.includes(filterId) || filterId === 'all'
+  )
+  const favoriteResults =
+    searchTerm === ''
+      ? favoriteResultsFiltered
+      : (searchDappList(favoriteResultsFiltered, searchTerm) as DappV2[])
+
+  useEffect(() => {
+    if (favoriteResults.length > 0) {
+      onShowSearchResult(false)
+    } else {
+      onShowSearchResult(true)
```

Comment:
i was confused about what `onShowSearchResult true` means...maybe this should just be `setFavoriteResultsEmpty`?

Reply:
Fixed in dd6b52699420a45ff16c1d098a54499930f379ea! This is also used in the search only variant, but I think a follow up pr aligning the behavior in those components would be better than adding more in this PR 😅